src/bootstrap/test.rs

```diff
@@ -2087,13 +2087,15 @@ impl Step for ErrorIndex {
         let mut tool = tool::ErrorIndex::command(builder);
         tool.arg("markdown").arg(&output);
 
-        let _guard =
-            builder.msg(Kind::Test, compiler.stage, "error-index", compiler.host, compiler.host);
-        let _time = util::timeit(&builder);
-        builder.run_quiet(&mut tool);
-        // The tests themselves need to link to std, so make sure it is
-        // available.
-        builder.ensure(compile::Std::new(compiler, compiler.host));
+        {
+            let _guard =
+                builder.msg(Kind::Test, compiler.stage, "error-index", compiler.host, compiler.host);
+            let _time = util::timeit(&builder);
+            builder.run_quiet(&mut tool);
+            // The tests themselves need to link to std, so make sure it is
+            // available.
+            builder.ensure(compile::Std::new(compiler, compiler.host));
```

Comment:
this needs to be moved out of the block to avoid nesting it

(probably it didn't panic because this is only relevant when cross-compiling? i'd expect x t error--index-generator --target i686 to break though.)